### PR TITLE
Adds config and communication to service docs

### PR DESCRIPTION
**Pull request by norling**
Adds service configuration information to the pipeline services.

### CODE REVIEW

**Review by blankdots — APPROVED** [review]

File: cmd/backup/backup.md [hunk 1]
```diff
@@ -2,6 +2,123 @@
 
 Moves data to backup storage and optionally merges it with the encryption header.
 
+## Configuration
+
+There are a number of options that can be set for the verify service.
```
Comment (aaperis) [inline cmd/backup/backup.md:7]:
```suggestion
There are a number of options that can be set for the backup service.
```

File: cmd/backup/backup.md [hunk 2]
```diff
@@ -2,6 +2,123 @@
...
+    - `allow`
+    - `prefer`
+    - `require`
+    - `verify-ca`
+    - `verify-full`
+
+   More information is available
+   [in the postgresql documentation](https://www.postgresql.org/docs/current/libpq-ssl.html#LIBPQ-SSL-PROTECTION)
+
+   Note that if `DB_SSLMODE` is set to anything by `disable`, then `DB_CACERT` needs to be set,
```
Comment (aaperis) [inline cmd/backup/backup.md:75]:
```suggestion
   Note that if `DB_SSLMODE` is set to anything but `disable`, then `DB_CACERT` needs to be set,
```

File: cmd/backup/backup.md [hunk 3]
```diff
@@ -2,6 +2,123 @@
...
+
+ - `DB_CACERT`: Certificate Authority (CA) certificate for the database to use
+
+### Storage settings
+
+Storage backend is defined by the `ARCHIVE_TYPE`, and `BACKUP_TYPE` variables.
+Valid values for these options are `S3` or `POSIX`
+(Defaults to `POSIX` on unknown values).
+
+The value of these variables define what other varaibles are read.
```
Comment (aaperis) [inline cmd/backup/backup.md:90]:
```suggestion
The value of these variables define what other variables are read.
```

File: cmd/finalize/finalize.md [hunk 4]
```diff
@@ -2,6 +2,87 @@
 
 Handles the so-called _Accession ID (stable ID)_ to filename mappings from Central EGA.
 
+
+## Configuration
+
+There are a number of options that can be set for the verify service.
```
Comment (aaperis) [inline cmd/finalize/finalize.md:8]:
```suggestion
There are a number of options that can be set for the finalize service.
```

File: cmd/intercept/intercept.md [hunk 5]
```diff
@@ -2,6 +2,84 @@
 
 The intercept service relays messages between Central EGA and Federated EGA nodes.
 
+## Configuration
+
+There are a number of options that can be set for the verify service.
```
Comment (aaperis) [inline cmd/intercept/intercept.md:7]:
```suggestion
There are a number of options that can be set for the intercept service.
```

File: cmd/finalize/finalize.md [hunk 6]
```diff
@@ -2,6 +2,87 @@
...
+    - `allow`
+    - `prefer`
+    - `require`
+    - `verify-ca`
+    - `verify-full`
+
+   More information is available
+   [in the postgresql documentation](https://www.postgresql.org/docs/current/libpq-ssl.html#LIBPQ-SSL-PROTECTION)
+
+   Note that if `DB_SSLMODE` is set to anything by `disable`, then `DB_CACERT` needs to be set,
```
Comment (aaperis) [inline cmd/finalize/finalize.md:63]:
```suggestion
   Note that if `DB_SSLMODE` is set to anything but `disable`, then `DB_CACERT` needs to be set,
```

File: cmd/ingest/ingest.md [hunk 7]
```diff
@@ -3,6 +3,117 @@
...
+    - `allow`
+    - `prefer`
+    - `require`
+    - `verify-ca`
+    - `verify-full`
+
+   More information is available
+   [in the postgresql documentation](https://www.postgresql.org/docs/current/libpq-ssl.html#LIBPQ-SSL-PROTECTION)
+
+   Note that if `DB_SSLMODE` is set to anything by `disable`, then `DB_CACERT` needs to be set,
```
Comment (aaperis) [inline cmd/ingest/ingest.md:70]:
```suggestion
   Note that if `DB_SSLMODE` is set to anything but `disable`, then `DB_CACERT` needs to be set,
```

File: cmd/intercept/intercept.md [hunk 8]
```diff
@@ -2,6 +2,84 @@
...
+    - `allow`
+    - `prefer`
+    - `require`
+    - `verify-ca`
+    - `verify-full`
+
+   More information is available
+   [in the postgresql documentation](https://www.postgresql.org/docs/current/libpq-ssl.html#LIBPQ-SSL-PROTECTION)
+
+   Note that if `DB_SSLMODE` is set to anything by `disable`, then `DB_CACERT` needs to be set,
```
Comment (aaperis) [inline cmd/intercept/intercept.md:60]:
```suggestion
   Note that if `DB_SSLMODE` is set to anything but `disable`, then `DB_CACERT` needs to be set,
```

File: cmd/mapper/mapper.md [hunk 9]
```diff
@@ -2,6 +2,84 @@
...
+    - `allow`
+    - `prefer`
+    - `require`
+    - `verify-ca`
+    - `verify-full`
+
+   More information is available
+   [in the postgresql documentation](https://www.postgresql.org/docs/current/libpq-ssl.html#LIBPQ-SSL-PROTECTION)
+
+   Note that if `DB_SSLMODE` is set to anything by `disable`, then `DB_CACERT` needs to be set,
```
Comment (aaperis) [inline cmd/mapper/mapper.md:60]:
```suggestion
   Note that if `DB_SSLMODE` is set to anything but `disable`, then `DB_CACERT` needs to be set,
```

File: cmd/verify/verify.md [hunk 10]
```diff
@@ -2,6 +2,117 @@
...
+    - `allow`
+    - `prefer`
+    - `require`
+    - `verify-ca`
+    - `verify-full`
+
+   More information is available
+   [in the postgresql documentation](https://www.postgresql.org/docs/current/libpq-ssl.html#LIBPQ-SSL-PROTECTION)
+
+   Note that if `DB_SSLMODE` is set to anything by `disable`, then `DB_CACERT` needs to be set,
```
Comment (aaperis) [inline cmd/verify/verify.md:69]:
```suggestion
   Note that if `DB_SSLMODE` is set to anything but `disable`, then `DB_CACERT` needs to be set,
```

File: cmd/mapper/mapper.md [hunk 11]
```diff
@@ -2,6 +2,84 @@
 
 The mapper service registers mapping of accessionIDs (stable ids for files) to datasetIDs.
 
+## Configuration
+
+There are a number of options that can be set for the verify service.
```
Comment (aaperis) [inline cmd/mapper/mapper.md:7]:
```suggestion
There are a number of options that can be set for the mapper service.
```

**Review by aaperis — COMMENTED** [review]
Looks good, except from some cp-paste typos :)

File: cmd/backup/backup.md [hunk 12]
```diff
@@ -2,6 +2,123 @@
...
+These settings control which crypt4gh keyfile is loaded.
+These settings are only needed is `copyheader` is `true`.
+
+ - `C4GH_FILEPATH`: path to the crypt4gh keyfile
+ - `C4GH_PASSPHRASE`: pass phrase to unlock the keyfile
+ - `C4GH_BACKUPPUBKEY`: path to the crypt4gh public key to use for reencrypting file headers.
+
+### RabbitMQ broker settings
+
+These settings control how ingest connects to the RabbitMQ message broker.
```
Comment (jbygdell) [inline cmd/backup/backup.md:37]:
```suggestion
These settings control how backup connects to the RabbitMQ message broker.
```

File: cmd/backup/backup.md [hunk 13]
```diff
@@ -2,6 +2,123 @@
...
+ - `*_URL`: URL to the S3 system
+ - `*_ACCESSKEY`: The S3 access and secret key are used to authenticate to S3,
+ [more info at AWS](https://docs.aws.amazon.com/general/latest/gr/aws-sec-cred-types.html#access-keys-and-secret-access-keys)
+ - `*_SECRETKEY`: The S3 access and secret key are used to authenticate to S3,
+ [more info at AWS](https://docs.aws.amazon.com/general/latest/gr/aws-sec-cred-types.html#access-keys-and-secret-access-keys)
+ - `*_BUCKET`: The S3 bucket to use as the storage root
+ - `*_PORT`: S3 connection port (default: `443`)
+ - `*_REGION`: S3 region (default: `us-east-1`)
+ - `*_CHUNKSIZE`: S3 chunk size for multipart uploads.
+ - `*_CACERT`: Certificate Authority (CA) certificate for the storage system
```
Comment (jbygdell) [inline cmd/backup/backup.md:103]:
```suggestion
# CA certificate is only needed if the S3 server has a certificate signed by a private entity
 - `*_CACERT`: Certificate Authority (CA) certificate for the storage system
```

File: cmd/finalize/finalize.md [hunk 14]
```diff
@@ -2,6 +2,87 @@
...
+```
+They may also be set using environment variables like:
+```bash
+export LOG_LEVEL="debug"
+export LOG_FORMAT="json"
+```
+
+### RabbitMQ broker settings
+
+These settings control how ingest connects to the RabbitMQ message broker.
```
Comment (jbygdell) [inline cmd/finalize/finalize.md:25]:
```suggestion
These settings control how finalize connects to the RabbitMQ message broker.
```

File: cmd/ingest/ingest.md [hunk 15]
```diff
@@ -3,6 +3,117 @@
...
+ - `*_URL`: URL to the S3 system
+ - `*_ACCESSKEY`: The S3 access and secret key are used to authenticate to S3,
+ [more info at AWS](https://docs.aws.amazon.com/general/latest/gr/aws-sec-cred-types.html#access-keys-and-secret-access-keys)
+ - `*_SECRETKEY`: The S3 access and secret key are used to authenticate to S3,
+ [more info at AWS](https://docs.aws.amazon.com/general/latest/gr/aws-sec-cred-types.html#access-keys-and-secret-access-keys)
+ - `*_BUCKET`: The S3 bucket to use as the storage root
+ - `*_PORT`: S3 connection port (default: `443`)
+ - `*_REGION`: S3 region (default: `us-east-1`)
+ - `*_CHUNKSIZE`: S3 chunk size for multipart uploads.
+ - `*_CACERT`: Certificate Authority (CA) certificate for the storage system
```
Comment (jbygdell) [inline cmd/ingest/ingest.md:98]:
```suggestion
# CA certificate is only needed if the S3 server has a certificate signed by a private entity
 - `*_CACERT`: Certificate Authority (CA) certificate for the storage system
```

File: cmd/finalize/finalize.md [hunk 16]
```diff
@@ -2,6 +2,87 @@
...
+
+ - `BROKER_HOST`: hostname of the rabbitmq server
+
+ - `BROKER_PORT`: rabbitmq broker port (commonly `5671` with TLS and `5672` without)
+
+ - `BROKER_QUEUE`: message queue to read messages from (commonly `accessionIDs`)
+
+ - `BROKER_ROUTINGKEY`: message queue to write success messages to (commonly `archived`)
```
Comment (jbygdell) [inline cmd/finalize/finalize.md:33]:
```suggestion
 - `BROKER_ROUTINGKEY`: message queue to write success messages to (commonly `completed`)
```

File: cmd/verify/verify.md [hunk 17]
```diff
@@ -2,6 +2,117 @@
...
+
+These settings control how ingest connects to the RabbitMQ message broker.
+
+ - `BROKER_HOST`: hostname of the rabbitmq server
+
+ - `BROKER_PORT`: rabbitmq broker port (commonly `5671` with TLS and `5672` without)
+
+ - `BROKER_QUEUE`: message queue to read messages from (commonly `archived`)
+
+ - `BROKER_ROUTINGKEY`: message queue to write success messages to (commonly `archived`)
```
Comment (jbygdell) [inline cmd/verify/verify.md:39]:
```suggestion
 - `BROKER_ROUTINGKEY`: message queue to write success messages to (commonly `verified`)
 # if `backup` is deployed the routing key should be `backup`
```

File: cmd/backup/backup.md [hunk 18]
```diff
@@ -2,6 +2,123 @@
...
+
+These settings control how ingest connects to the RabbitMQ message broker.
+
+ - `BROKER_HOST`: hostname of the rabbitmq server
+
+ - `BROKER_PORT`: rabbitmq broker port (commonly `5671` with TLS and `5672` without)
+
+ - `BROKER_QUEUE`: message queue to read messages from (commonly `backup`)
+
+ - `BROKER_ROUTINGKEY`: message queue to write success messages to (commonly `archived`)
```
Comment (jbygdell) [inline cmd/backup/backup.md:45]:
```suggestion
 - `BROKER_ROUTINGKEY`: message queue to write success messages to (commonly `verified`)
```

File: cmd/verify/verify.md [hunk 19]
```diff
@@ -2,6 +2,117 @@
...
+ - `*_URL`: URL to the S3 system
+ - `*_ACCESSKEY`: The S3 access and secret key are used to authenticate to S3,
+ [more info at AWS](https://docs.aws.amazon.com/general/latest/gr/aws-sec-cred-types.html#access-keys-and-secret-access-keys)
+ - `*_SECRETKEY`: The S3 access and secret key are used to authenticate to S3,
+ [more info at AWS](https://docs.aws.amazon.com/general/latest/gr/aws-sec-cred-types.html#access-keys-and-secret-access-keys)
+ - `*_BUCKET`: The S3 bucket to use as the storage root
+ - `*_PORT`: S3 connection port (default: `443`)
+ - `*_REGION`: S3 region (default: `us-east-1`)
+ - `*_CHUNKSIZE`: S3 chunk size for multipart uploads.
+ - `*_CACERT`: Certificate Authority (CA) certificate for the storage system
```
Comment (jbygdell) [inline cmd/verify/verify.md:97]:
```suggestion
# CA certificate is only needed if the S3 server has a certificate signed by a private entity
 - `*_CACERT`: Certificate Authority (CA) certificate for the storage system
```

File: cmd/intercept/intercept.md [hunk 20]
```diff
@@ -2,6 +2,84 @@
...
+```
+They may also be set using environment variables like:
+```bash
+export LOG_LEVEL="debug"
+export LOG_FORMAT="json"
+```
+
+### RabbitMQ broker settings
+
+These settings control how ingest connects to the RabbitMQ message broker.
```
Comment (jbygdell) [inline cmd/intercept/intercept.md:24]:
```suggestion
These settings control how intercept connects to the RabbitMQ message broker.
```

File: cmd/mapper/mapper.md [hunk 21]
```diff
@@ -2,6 +2,84 @@
...
+```
+They may also be set using environment variables like:
+```bash
+export LOG_LEVEL="debug"
+export LOG_FORMAT="json"
+```
+
+### RabbitMQ broker settings
+
+These settings control how ingest connects to the RabbitMQ message broker.
```
Comment (jbygdell) [inline cmd/mapper/mapper.md:24]:
```suggestion
These settings control how mapper connects to the RabbitMQ message broker.
```

**Review by jbygdell — DISMISSED** [review]

**Review by kostas-kou — APPROVED** [review]
Looks good to me and I agree with the comments of the other reviewers

**Review by kjellp — APPROVED** [review]
Looks good.

Seems like the default routingkey setting docs now indicate that the backup service should come after ingest and before verify? The overview schema on the front-page indicates ingest->verify->backup->finalize, so this is not fully in sync now.

**Comment by norling** [comment]
I honestly don't know the "intended" order of backup :D if anyone knows - please update with some comment!

File: cmd/finalize/finalize.md [hunk 22]
```diff
@@ -2,6 +2,87 @@
...
+
+These settings control how finalize connects to the RabbitMQ message broker.
+
+ - `BROKER_HOST`: hostname of the rabbitmq server
+
+ - `BROKER_PORT`: rabbitmq broker port (commonly `5671` with TLS and `5672` without)
+
+ - `BROKER_QUEUE`: message queue to read messages from (commonly `accessionIDs`)
+
+ - `BROKER_ROUTINGKEY`: message queue to write success messages to (commonly `completed`)
```
Comment (blankdots) [inline cmd/finalize/finalize.md:33]:
```suggestion
 - `BROKER_ROUTINGKEY`: message queue to write success messages to (commonly `backup`)
```

File: cmd/verify/verify.md [hunk 23]
```diff
@@ -2,6 +2,119 @@
...
+These settings control how verify connects to the RabbitMQ message broker.
+
+ - `BROKER_HOST`: hostname of the rabbitmq server
+
+ - `BROKER_PORT`: rabbitmq broker port (commonly `5671` with TLS and `5672` without)
+
+ - `BROKER_QUEUE`: message queue to read messages from (commonly `archived`)
+
+ - `BROKER_ROUTINGKEY`: message queue to write success messages to (commonly `verified`)
+ # if `backup` is deployed the routing key should be `backup`
```
Comment (blankdots) [inline cmd/verify/verify.md:40]:
```suggestion
```

**Review by blankdots — CHANGES_REQUESTED** [review]
following the sda-helm deployment

i think the flow is  `ingest`->`verify`->`finalize`-> `backup` and ->`mapper` happens at some point after finalize

**Review by blankdots — APPROVED** [review]

File: cmd/mapper/mapper.md [hunk 24]
```diff
@@ -16,3 +94,10 @@ If the message can’t be validated it is discarded with an error message in the
 On error the service sleeps for up to 5 minutes to allow for database recovery, after 5 minutes the message is Nacked, re-queued and an error message is written to the logs.
 
 1. The RabbitMQ message is Ack'ed.
+
+
+## Communication
+
+ - Mapper reads messages from one rabbitmq queue (default `mappings`).
+
+ - Mapper maps files to datasets in the database using thre `MapFilesToDataset` function.
```
Comment (pontus) [inline cmd/mapper/mapper.md:103]:
```suggestion
 - Mapper maps files to datasets in the database using the `MapFilesToDataset` function.
```

**Review by pontus — APPROVED** [review]

**Comment by pontus** [comment]
I think this is a clear improvement and don't see any issues that should block merging.

Possibly, one might consider moving all common configuration to one place rather than having it repeated all over (and risk missing updating things somewhere when adding new).

**Pull request merged**